pkg/background/common/labels.go

```diff
@@ -51,17 +51,17 @@ func MutateLabelsSet(policyKey string, trigger Object) pkglabels.Set {
 	return set
 }
 
-func GenerateLabelsSet(policyKey string, trigger Object) pkglabels.Set {
+func GenerateLabelsSet(policyKey string, trigger Object, triggerResourceUID string) pkglabels.Set {
```

Comment:
But it would mean introducing a new field `UID` in `ResourceSpec` (https://github.com/kyverno/kyverno/blob/main/api/kyverno/v1/resource_spec_types.go#L11-L23) because `trigger` is of type `kyvernov1.ResourceSpec` https://github.com/kyverno/kyverno/blob/5720df780c8be8a7a18b84fcc62e401ca305542d/pkg/policy/updaterequest.go#L13-L27 

Should a new field be introduced here?

Reply:
Yes.